src/lib/vip-import-validate-files.js

```diff
@@ -0,0 +1,366 @@
+	let errors = 0;
+	let yearIndex, monthIndex;
+
+	// Turn the path into an array to determine index position
+	const directories = folderStructure.split( '/' );
+
+	/**
+	 * Upload folder validation
+	 *
+	 * Find if an `uploads` folder exists and return its index position
+	 */
+	const uploadsIndex = directories.indexOf( 'uploads' );
+
+	/**
+	 * Year folder validation
+	 *
+	 * Find if a year folder exists via a four digit regex matching pattern,
+	 * then obtain that value
+	 */
+	const regexYear = /\b\d{4}\b/g;
+	const year = regexYear.exec( folderStructure ); // Returns an array with the regex-matching value
+
+	if ( year ) {
+		yearIndex = directories.indexOf( year[ 0 ] );
+	}
+
+	/**
+	 * Month folder validation
+	 *
+	 * Find if a month folder exists via a two digit regex matching pattern,
+	 * then obtain that value
+	 */
+	const regexMonth = /\b\d{2}\b/g;
+	const month = regexMonth.exec( folderStructure ); // Returns an array with the regex-matching value
+
+	if ( month ) {
+		monthIndex = directories.indexOf( month[ 0 ] );
+	}
+
+	/**
+	 * Logging
+	 */
+
+	// Uploads folder
+	if ( uploadsIndex === 0 ) {
+		console.log();
+		console.log( '✅ File structure: Uploads directory exists' );
+		console.log();
+	} else {
+		console.log();
+		console.log( chalk.yellow( '✕' ), 'Recommended: Media files should reside in an', chalk.magenta( '`uploads`' ), 'directory' );
+		errors++;
+	}
+
+	// Year folder
+	if ( yearIndex && yearIndex === 1 ) {
+		console.log( '✅ File structure: Year directory exists (format: YYYY)' );
+		console.log();
+	} else {
+		console.log( chalk.yellow( '✕' ), 'Recommended: Structure your WordPress media files into', chalk.magenta( '`uploads/YYYY`' ), 'directories' );
+		errors++;
+	}
+
+	// Month folder
+	if ( monthIndex && monthIndex === 2 ) {
+		console.log( '✅ File structure: Month directory exists (format: MM)' );
+		console.log();
+	} else {
+		console.log( chalk.yellow( '✕' ), 'Recommended: Structure your WordPress media files into', chalk.magenta( '`uploads/YYYY/MM`' ), 'directories' );
+		console.log();
+		errors++;
+	}
+
+	if ( errors ) {
+		recommendedFileStructure();
+	}
+
+	return;
+};
+
+/** Character validation
+ *
+ * This logic is based on the WordPress core function `sanitize_file_name()`
+ * https://developer.wordpress.org/reference/functions/sanitize_file_name/
+ *
+ * @param {string} filename - The current file being validated
+ * @returns {Boolean} - Checks if the filename has been sanitized
+ */
+export const isFileSanitized = filename => {
+	let sanitizedFile = filename;
+
+	// Prohibited characters:
+	// Encoded spaces (%20), no-break spaces - keeps words together (\u00A0), and plus signs
+	const regexSpaces = /\u00A0|(%20)|\+/g;
+	sanitizedFile = sanitizedFile.replace( regexSpaces, ' ' );
+
+	// Prohibited characters:
+	// Special characters: + & # % = ' " \ < > : ; , / ? $ * | ` ! { }
+	const regexSpecialChars = /[\/\'\"\\=<>:;,&?$#*|`!+{}%]/g;
```

Comment:
See earlier comment about storing these in one place.  It would be great in the event that it needs to be updated to have a single source of truth 👍

Reply:
Moved `regexSpecialChars` to a global variable in 98c18f5!

I didn't change it to a `Set` because the regex order is important and I believe `Set`s are unordered.

Changing it to an array seemed a bit excessive, accessing the characters as `regexSpecialChars[ 0 ]` and such, but please let me know if I'm mistaken or there's a better execution! Thank you!